### PR TITLE
{Compute} Detach Azure Stack from Profiles 

### PR DESCRIPTION
**Pull request by yanzhudd**
**Related command**

**Description**

Brief procedures:
- add command loader for azure stack
- build azure stack context

The following files/folder are still used in common:
- _alias.py
- _arg_client.py
- _workspace_data_source_settings.py
- _vmss_application_health.py
- _help.py
- linter_exclusions.yml
- generated folder
- manual folder

aaz PR: https://github.com/Azure/aaz/pull/633

**Testing Guide**

1. `az cloud set --name AzureCloud --profile 2018-03-01-hybrid`
2. `azdev test vm --no-exitfirst`

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -1528,13 +1536,29 @@ def load_arguments(self, _):
...
+        c.argument('proximity_placement_group', options_list=['--ppg'], help="The name or ID of the proximity placement group the VM should be associated with.",
+                   validator=_validate_proximity_placement_group)
+
+    with self.argument_context('vmss update', min_api='2018-04-01') as c:
+        c.argument('proximity_placement_group', options_list=['--ppg'], help="The name or ID of the proximity placement group the VMSS should be associated with.",
+                   validator=_validate_proximity_placement_group)
+
+    with self.argument_context('vm availability-set update', min_api='2018-04-01') as c:
+        c.argument('proximity_placement_group', options_list=['--ppg'], help="The name or ID of the proximity placement group the availability set should be associated with.",
+                   validator=_validate_proximity_placement_group)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:1561]: Why do we need to refactor this piece of code?

File: src/azure-cli/azure/cli/command_modules/vm/aaz/profile_2020_09_01_hybrid/capacity/reservation/group/_list.py [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+@register_command(
+    "capacity reservation group list",
+)
+class List(AAZCommand):
+    """List the capacity reservation groups.
+
+    :example: List capacity reservation groups
+        az capacity reservation group list -g rg
+
+    :example: List the capacity reservation groups containing VM instances and VMSS instance which are     associated to capacity reservation group
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/aaz/profile_2020_09_01_hybrid/capacity/reservation/group/_list.py:23]:
```suggestion
    :example: List the capacity reservation groups containing VM instances and VMSS instance which are associated to capacity reservation group
```

File: src/azure-cli/azure/cli/command_modules/vm/azure_stack/commands.py [hunk 3]
```diff
@@ -0,0 +1,603 @@
...
+        g.custom_show_command('show', 'restore_point_show')
+        g.custom_command('create', 'restore_point_create', supports_no_wait=True)
+        g.wait_command('wait')
+
+    with self.command_group('restore-point collection', restore_point_collection, min_api='2021-03-01',
+                            client_factory=cf_restore_point_collection) as g:
+        g.custom_show_command('show', 'restore_point_collection_show')
+        g.custom_command('create', 'restore_point_collection_create')
+        g.custom_command('update', 'restore_point_collection_update')
+        g.wait_command('wait')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/azure_stack/commands.py:603]: > min_api='2021-03-01'

Please remove these code unrelated to Azure Stack in future PRs

File: src/azure-cli/azure/cli/command_modules/vm/azure_stack/_constants.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+# license information.
+#
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is
+# regenerated.
+#
+# Generation mode: Incremental
+# --------------------------------------------------------------------------
+
+UPGRADE_SECURITY_HINT = 'Consider upgrading security for your workloads using Azure Trusted Launch VMs. ' \
```
[the remark this answers is not in the document]
Comment (yanzhudd) [reply src/azure-cli/azure/cli/command_modules/vm/azure_stack/_constants.py:13]: We may not need it anymore, but it is not related to this PR. Will fix it in a future PR.

File: src/azure-cli/azure/cli/command_modules/vm/operations/vmss.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    def _output(self, *args, **kwargs):
+        from azure.cli.core.aaz import AAZUndefined, has_value
+
+        # Resolve flatten conflict
+        # When the type field conflicts, the type in inner layer is ignored and the outer layer is applied
+        for value in self.ctx.vars.instance.value:
+            if has_value(value.resources):
+                for resource in value.resources:
+                    if has_value(resource.type):
+                        resource.type = AAZUndefined
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/operations/vmss.py:23]: Why do we need this logic?

File: src/azure-cli/azure/cli/command_modules/vm/azure_stack/commands.py (hunk 3 above)
Comment (yanzhudd) [reply src/azure-cli/azure/cli/command_modules/vm/azure_stack/commands.py:603]: will remove them in a future migration PR

File: src/azure-cli/azure/cli/command_modules/vm/operations/vmss.py (hunk 5 above)
Comment (yanzhudd) [reply src/azure-cli/azure/cli/command_modules/vm/operations/vmss.py:23]: It is moved from custom.py file and not related to this PR. It was imported by this PR: https://github.com/Azure/azure-cli/pull/27128
I remembered that this was caused by a conflict in the output field in swagger. Had a discussion with Kai at the time and this was the best solution.

File: src/azure-cli/azure/cli/command_modules/vm/aaz/profile_2020_09_01_hybrid/capacity/reservation/group/_list.py (hunk 2 above)
Comment (yanzhudd) [reply src/azure-cli/azure/cli/command_modules/vm/aaz/profile_2020_09_01_hybrid/capacity/reservation/group/_list.py]: redundant spaces removed

File: src/azure-cli/azure/cli/command_modules/vm/azure_stack/commands.py (hunk 3 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/vm/azure_stack/commands.py:603]: OK, sounds good

**Pull request merged**